GoogleMapsComponents/wwwroot/js/objectManager.js

```diff
@@ -463,6 +500,20 @@
                         delete mapObjects[elementToRemove];
                     }
                 }
+
+                if(controlParents.hasOwnProperty(mapGuid)){
+                    for (var key in controlParents[mapGuid]) {
+                        if (controlParents[mapGuid].hasOwnProperty(key)) {
+                            var parent = controlParents[mapGuid][key];
+                            if(!parent) continue
+                            var elem = document.getElementById(key);
+                            if(!elem) continue
+                            parent.appendChild(elem);
+                            elem.style.display = "none";
+                        }
+                    }
+                    delete controlParents[mapGuid];
```

Comment:
My knowleadge about js is very limited. But maybe check if controlParents dont have anything and set to null. I am afraid that we could have hanging smth here and consuming resources.
Also please check dispose  case when you have two maps with multiple controls

Reply:
Yes will check

Reply:
setting controlParents to null if empty.

Tested multiple maps and dispose and it works fine.